x-pack/plugins/alerts/server/saved_objects/index.ts

```diff
@@ -9,6 +9,19 @@ import mappings from './mappings.json';
 import { getMigrations } from './migrations';
 import { EncryptedSavedObjectsPluginSetup } from '../../../encrypted_saved_objects/server';
 
+export { partiallyUpdateAlert } from './partially_update_alert';
+
+export const AlertAttributesExcludedFromAAD = [
+  'scheduledTaskId',
+  'muteAll',
+  'mutedInstanceIds',
+  'updatedBy',
+];
```

Comment:
I don't think this is doing what you had in mind, the type signature looks wrong.

I suspect you meant this though:
```suggestion
export const AlertAttributesExcludedFromAAD = [
  'scheduledTaskId',
  'muteAll',
  'mutedInstanceIds',
  'updatedBy',
] as const;
```

Reply:
Huh.  Any chance this is a case of vscode being confused vs a typescript error.  I'll take another look, though, see if it's actually allowing other attrs in.

I copy pasta'd this from an example I had seen about a week ago - if it's too obscure anyway (I'd say it's fairly obscure), I can switch this back to the explicit verison where we list the attr names twice - it's not too bad with just 4 attrs.

Reply:
we could use enums here, or change to use the `|` syntax

Reply:
I changed from the confusing 

```
export type AlertAttributesExcludedFromAADType = typeof AlertAttributesExcludedFromAAD[number];
```

to the more verbose, but clearer:

```
export type AlertAttributesExcludedFromAADType =
  | 'scheduledTaskId'
  | 'muteAll'
  | 'mutedInstanceIds'
  | 'updatedBy';
```

Both the attribute names as an array of strings, and this type definition, are right next to each other in the code, so hopefully it will be simple to keep these in sync if they need to be changed in the future.